### PR TITLE
GEOMESA-2979 Pass QUERY_INDEX hint to stats and getCount

### PR DESCRIPTION
**Pull request by jnh5y**
Signed-off-by: Jim Hughes <jhughes@ccri.com>

### CODE REVIEW

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/stats/MetadataBackedStats.scala [hunk 1]
```diff
@@ -37,9 +38,10 @@ trait MetadataBackedStats extends GeoMesaStats with StatsBasedEstimator with Laz
   protected def ds: HasGeoMesaMetadata[String]
   protected def generateStats: Boolean
 
-  override def getCount(sft: SimpleFeatureType, filter: Filter, exact: Boolean): Option[Long] = {
+  override def getCount(sft: SimpleFeatureType, filter: Filter, exact: Boolean, queryHints: Hints): Option[Long] = {
     if (exact) {
-      runStats[CountStat](sft, Stat.Count(), filter).headOption.map(_.count)
+      // TODO: runStats needs to be updated!
```
Comment (jnh5y) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/stats/MetadataBackedStats.scala:43]:
```suggestion
```

File: geomesa-index-api/src/test/resources/log4j.xml [hunk 2]
```diff
@@ -23,11 +23,11 @@
     <!-- un-comment the following line to enable verbose log messages
          from the index query-planner; this can be helpful in debugging
          query plans -->
-    <!--
+    <!-- -->
```
Comment (jnh5y) [inline geomesa-index-api/src/test/resources/log4j.xml:26]:
```suggestion
    <!-- 
```

File: geomesa-index-api/src/test/resources/log4j.xml [hunk 3]
```diff
@@ -23,11 +23,11 @@
...
     <category name="org.locationtech.geomesa.index.utils.Explainer">
         <priority value="trace"/>
     </category>
-    -->
+    <!-- -->
```
Comment (jnh5y) [inline geomesa-index-api/src/test/resources/log4j.xml:30]:
```suggestion
    -->
```

**Comment by elahrvivaz** [comment]
build is dying due to gt 15 artifacts no longer being online... @jnh5y did you build it locally?

**Comment by jnh5y** [comment]
Let me kick off a local build (which will take awhile).  I also kinda want to here from @michaelchehreh-ccri that he's happy with it:).

**Comment by jnh5y** [comment]
My local build just finished successfully!

**Pull request merged**